### ISSUE TITLE
xmlstarlet not installed

### ISSUE THREAD

**Issue by ceagan** [jboss-dockerfiles/wildfly#41]
In the switch from Fedora to CentOS as the base, the base effectively lost xmlstarlet. On CentOS, xmlstarlet must be installed through the EPEL repositories.

I opened this issue (https://github.com/jboss-dockerfiles/base/issues/4) awhile back, but it hasn't received any attention. I am referencing it here because the issue affects users of the Wildfly Docker instance and xmlstarlet is referenced in the documentation for configuring Wildfly in Docker.

Until the base version is updated, I created a workaround (https://github.com/jboss-dockerfiles/wildfly) which resolves the problem. It is on Docker Hub (https://hub.docker.com/r/ceagan/wildfly/) as well.

**Comment by goldmann** [comment]
Hey, what documentation are you talking about? Are you thinking about my blog post? https://goldmann.pl/blog/2014/07/23/customizing-the-configuration-of-the-wildfly-docker-image/

**Comment by ceagan** [comment]
Yes, that's the one I was remembering. I realize that it is getting a bit old now. Is there a better way being recommended these days to solve the issues? I use a combination of xmlstarlet and jboss-cli to perform our initial configurations.

**Comment by goldmann** [comment]
There is no single solution. Even now, two years after this blog post -it is pretty accurate and summary is still valid.

The problem is that with single, general purpose image we cannot (and we don't try to) satisfy everyone. This may look bad from one side, but from the other side maintenance costs are just too high. Additionally since Docker makes it so easy to base on our image - there is no reason why upstream image should fit all needs.

But, since XML is popular in JBoss Middleware and xmlstarlet is a pretty easy way to modify it - I'll add it (back) to the base image.

**Comment by ceagan** [comment]
I agree that the inclusion of additional software in the base image is not the best plan from a maintenance perspective and it also bloats the images for users who don't need the additional functionality.

I think that the bug is valid in this case because the base image still includes the code that tries to install xmlstarlet. I think that either removing the code from the Dockerfile that attempts to install xmlstarlet or fixing it so that the package installs properly would resolve the issue. I remember being very confused when I was trying to use the image the first time because it looked like the Dockerfile was installing xmlstarlet, but I couldn't get it to work.

Thanks for replying and providing a solution so that other users won't face the same confusion I did.

**Comment by loxy** [comment]
Still not in it...
